src/kpro.erl

```diff
@@ -450,18 +453,28 @@ decode(StructName, Bin) when is_atom(StructName) ->
   kpro_structs:decode(StructName, Bin).
 
 %% @private
+-spec decode_message(binary()) ->
+        {kpro_Message() | incomplete_message(), binary()}.
+decode_message(<<_Offset:64/?INT, MsgSize:32/?INT, T/binary>> = Bin) ->
+  case size(T) < MsgSize of
+    true  -> {{?incomplete_message, MsgSize + 12}, <<>>};
+    false -> decode(kpro_Message, Bin)
+  end;
+decode_message(_) ->
+  %% need to fetch at least 12 bytes to know the message size
```

Comment:
on disk, there is 8 bytes of offset then 4 bytes of message-size followed by the message body.
the message body has crc magic-byte and all that, but included in the message-size.